src/shiftedNormL1B2.jl

```diff
@@ -47,17 +47,15 @@ function prox(
   q::AbstractVector{R},
   σ::R,
 ) where {R <: Real, V0 <: AbstractVector{R}, V1 <: AbstractVector{R}, V2 <: AbstractVector{R}}
-  ProjB(y) = min.(max.(y, q .- ψ.λ * σ), q .+ ψ.λ * σ)
-  froot(η) = η - ψ.χ(ProjB((-ψ.sj - ψ.xk) .* (η / ψ.Δ)))
+  ProjB(y) = min.(max.(y, ψ.sj .+ q .- ψ.λ * σ), ψ.sj .+ q .+ ψ.λ * σ)
+  froot(η) = η - ψ.χ(ProjB((- ψ.xk) .* (η / ψ.Δ)))
 
-  ψ.sol .= ProjB(-ψ.sj - ψ.xk)
+  ψ.sol .= ProjB(- ψ.xk)
 
   if ψ.χ(ψ.sol) > ψ.Δ
     η = fzero(froot, 1e-10, Inf)
-    ψ.sol .*= (η / ψ.Δ)
-  end
-  if ψ.χ(ψ.sol) > ψ.Δ
-    ψ.sol .*= (ψ.Δ / ψ.χ(ψ.sol))
+    ψ.sol .= ProjB((- ψ.xk) .* (η / ψ.Δ)) * (ψ.Δ / η)
```

Comment:
Maybe recompute the norm here instead of reusing η? Currently, you’re not exactly on ΔB2 because you didn’t solve the equation exactly.

Reply:
hmm actually i'm not sure if i understand you here. You mean use η in the ProjB() function but then use the norm of ψ.sol in the outer scaling?

Reply:
Yes. Otherwise, you won’t be exactly on the boundary.

Reply:
Any success with this?

Reply:
Not really - the root-finding itself is producing the nan's, and retaking the norm of ψ.sol really slows down the procedure.

Reply:
Were you able to output the iteration detail of the root finder to try and find out what's going wrong?

Reply:
pushed a new commit where the error doesn't seem to crop up. The fzero method is different in Roots.jl than matlab, and seems to shoot off towards Inf when our algorithm approaches convergence. "find_zero()" does not do it, and I think initializing at Delta helps.